tests/cpp_validation_tests/test_validation.cpp

```diff
@@ -132,16 +140,20 @@ OwningDataset create_result_dataset(OwningDataset const& input, std::string cons
                                     Idx batch_size = 1) {
     OwningDataset owning_dataset{.dataset{DatasetMutable{dataset_name, is_batch, batch_size}},
                                  .const_dataset = std::nullopt};
-    auto const& info_input = input.const_dataset.value().get_info();
 
-    for (Idx component_idx{}; component_idx != info_input.n_components(); ++component_idx) {
-        auto const& component_name = info_input.component_name(component_idx);
+    if (!input.const_dataset.has_value()) {
+        throw OptionalNotInitialized("DatasetConst");
```

Comment:
here and below `UnreachableHit` could also do the trick but it's fine to keep as is

Reply:
I see that `UnreachableHit` lives in `power_grid_model` only, so I will keep as is for now. If error management becomes an issue later, we can open an issue when/if that happens.

Reply:
OK